src/OptionsPopup.qml

```diff
@@ -404,15 +404,14 @@ Window {
                     // textFormat: Text.PlainText
                     Layout.leftMargin: 40
                 }
-                TextArea {
+                TextField {
                     id: fieldPublicKey
                     enabled: radioPubKeyAuthentication.checked
-                    textFormat: TextEdit.PlainText
                     wrapMode: TextEdit.WrapAnywhere
                     Layout.fillWidth: true
                     Layout.minimumWidth: 350
                     Layout.leftMargin: 40
-                    selectByMouse: true
+                    validator: RegularExpressionValidator { regularExpression: /^ssh-(ed25519|rsa|dss|ecdsa) AAAA(?:[A-Za-z0-9+\/]{4})*(?:[A-Za-z0-9+\/]{2}==|[A-Za-z0-9+\/]{3}=|[A-Za-z0-9+\/]{4})( [^@]+@[^@]+)?/ }
```

Comment:
If the aim of this PR is to fix #956 by preventing `'` or `"` from  appearing in the public key field (which breaks `firstrun.sh`), perhaps the use of `[^@]+` is a bit overly broad? :thinking:

Reply:
> If the aim of this PR is to fix #956 by preventing `'` or `"` from appearing in the public key field (which breaks `firstrun.sh`), perhaps the use of `[^@]+` is a bit overly broad? 🤔

You're correct - this would prevent one appearing at the _start_ of the ssh-key, but not at the end.